Cross compile for 2.11.7, RegExValidator and InCaseValidator

EdgeCaseBerg opened the pull request.

EdgeCaseBerg committed:
Update scalatest dependency, cross compile

Add cross compilation for 2.11.7 and update scalatest to a version that
works for both the old and the new scala versions. Running +compile and
+test both work fine.

EdgeCaseBerg committed:
Add RegexValidator and test

Using re.findAllIn because unapplySeq doesn't seem to behave as I'd
expect it to.

EdgeCaseBerg committed:
Use RegexValidator in SimpleEmailValidator

We can't extend a case class from another one though or scala will
complain, so instead we use the regex validator directly in the isValid
method.

EdgeCaseBerg committed:
Add InCaseValidator and test

Add a simple predicate validator as well as a small scalatest to make
sure it works.

siman commented on src/main/scala/com/plameno/validation/CommonValidators.scala:
```diff
@@ -23,12 +25,16 @@ case class MaxLengthValidator(s: String, maxLen: Int, customError: Option[String
-// TODO: RegexValidator with compiled pattern. And reuse it for email validation.
+case class RegexValidator(s: String, re: Regex, customError: Option[String] = None)
+  extends Validator(customError) {
+  override val defaultError = s"Regular expression ${re} could not be matched"
```
Surround ${re} with quotes '' to have more user friendly message:
`s"Regular expression '${re}' could not`...

EdgeCaseBerg replied:
Good call. Addressed in a1f3e05

siman commented on src/test/scala/com/plameno/validation/ValidationSpec.scala:
```diff
@@ -32,6 +32,17 @@ class ValidationSpec extends FlatSpec {
+  }
+
+  it should "validate any supplied function in the InCaseValidator" in {
+    assert(InCaseValidator(2, (x : Int) => {x % 2 == 0}).isValid)
```
Here should be 2 test cases:
1. valid (you already made it)
2. invalid (TODO)

EdgeCaseBerg replied:
Good point.
addressed in cb40ca7

siman commented on project/Build.scala:
```diff
@@ -12,6 +12,7 @@ object BuildSettings {
     organization := "com.plameno",
     version := "0.1-SNAPSHOT",
     scalaVersion := "2.10.2",
+    crossScalaVersions := Seq("2.11.7"),
```
:cool:

EdgeCaseBerg replied:
Isn't it? :) 
You can test both versions by adding the `+` sign before an SBT command.

EdgeCaseBerg committed:
Make RegexValidator error message more friendly

Easier to parse the message on what the regular expression was with
qoutes as suggested by siman

EdgeCaseBerg committed:
Simplify errorMessage value in test

EdgeCaseBerg committed:
Add invalid test for InCaseValidator

siman commented:
Good job! Just wondering where do you use these validations?

EdgeCaseBerg commented:
I was looking for a simple scala validator library last night and stumbled across yours. I had dealt with a [different validation library](https://github.com/wix/accord) before that was a bit more complicated  and promised custom error messages, but I couldn't get the error messaging to work. 

Your code is way easier to understand than the accord stuff and I noticed the To-do's and thought I might contribute a little :)
